cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -869,19 +882,21 @@ private NamespaceId validateNamespace(String namespaceId) throws BadRequestExcep
       // Hence, this exception will never be thrown
       throw Throwables.propagate(e);
     }
-    return namespace;
+    return namespaceId;
   }
 
-  private ApplicationId validateApplicationId(String namespace, String appId)
+  private void validateApplicationId(@Nullable String namespace, @Nullable String appId)
     throws BadRequestException, NamespaceNotFoundException, AccessException {
-    return validateApplicationId(validateNamespace(namespace), appId);
+    validateApplicationId(validateNamespace(namespace), appId);
   }
 
-  private ApplicationId validateApplicationId(NamespaceId namespaceId, String appId) throws BadRequestException {
-    return validateApplicationVersionId(namespaceId, appId, ApplicationId.DEFAULT_VERSION);
+  private void  validateApplicationId(NamespaceId namespaceId, String appId) throws BadRequestException {
+    validateApplicationVersionId(namespaceId, appId, ApplicationId.DEFAULT_VERSION);
```

Comment:
It should be visa-versa. You are doing extra work for no reason. This call should validate appId and `validateApplicationVersionId` should call this method and then validate version

Reply:
This will be part of the cleanup and refactor for https://cdap.atlassian.net/browse/CDAP-19937